add HCP for Z (Editing/rewriting in process) (#5647)
open-cluster-management/rhacm-docs: clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc

[#configuring-hosting-service-cluster-ibmz]
= Configuring the hosting cluster on `x86` bare metal for IBM Z compute nodes (Technology Preview)

> **Technology Preview:** Configuring the hosting cluster on 'x86' bare metal is in a Technology Preview status with limited support.

Reviewer: On line 4, how are we intending to format "x86"? Should it be in monospace, as it is in the heading? If so, replace the single quotation marks with back ticks.

You can deploy hosted control planes by configuring a cluster to function as a hosting cluster. The hosting cluster is the {ocp-short} cluster where the control planes are hosted. The hosting cluster is also known as the _management_ cluster. 

*Note:* The _management_ cluster is not the _managed_ cluster. A managed cluster is a cluster that the hub cluster manages.

> You can convert a managed cluster to a hosting cluster by using the `hypershift` add-on to deploy the HyperShift Operator on that cluster. Then, you can start to create the hosted cluster. 

Reviewer: Is the statement on line 10 still true? I thought that since the HyperShift feature is now enabled by default, you don't have to explicitly do anything with the add-on anymore.

The {mce-short} 2.4 supports only the default `local-cluster`, which is a hub cluster that is managed, and the hub cluster as the hosting cluster.

> *Note:* IBM Z does not support ISO boot.

Reviewer: Instead of placing two admonitions side by side, I suggest moving "IBM Z does not support ISO boot." to be the first bullet in the "Important" list.

Author: Done

**Important:**

> - To provision hosted control planes on bare metal, you can use the Agent platform. The Agent platform uses the Central Infrastructure Management (CIM) service to add worker nodes to a hosted cluster. For an introduction to the CIM service, see link:https://github.com/openshift/assisted-service/blob/master/docs/hive-integration/kube-api-getting-started.md[Kube API - Getting Started Guide].

Reviewer: ```suggestion
- To provision hosted control planes on bare metal, you can use the Agent platform. The Agent platform uses the central infrastructure management service to add worker nodes to a hosted cluster. For an introduction to the central infrastructure management service, see link:https://github.com/openshift/assisted-service/blob/master/docs/hive-integration/kube-api-getting-started.md[Kube API - Getting Started Guide].
```

Author: Done

> - Each IBM Z system host must be started with the PXE images provided by the CIM. After each host starts, it runs an Agent process to discover the details of the host and completes the installation. An Agent custom resource represents each host.

Reviewer: ```suggestion
- Each IBM Z system host must be started with the PXE images provided by the central infrastructure management service. After each host starts, it runs an Agent process to discover the details of the host and completes the installation. An Agent custom resource represents each host.
```

Author: Done

> - When you create a hosted cluster with the Agent platform, HyperShift installs the Agent Cluster API provider in the hosted control plane namespace.

Reviewer: On line 22, what do you mean by "HyperShift"? Are you referring to the HyperShift Operator? If so, please use "HyperShift Operator". If you're referring to the hosted control planes feature, use "hosted control planes".

- When you scale down a node pool, Agents are unbound from the corresponding cluster. Before you reuse the clusters, you must boot the clusters by using the PXE image to update the number of nodes.

[#hosting-service-cluster-configure-prereq-ibmz]
== Prerequisites

* {mce} version 2.4 or later must be installed on an {ocp-short} cluster. The {mce-short} is installed automatically when you install {product-title-short}. You can also install {mce-short} without {product-title-short} as an Operator from the {ocp-short} OperatorHub.

* You need the {mce-short} must have at least one managed {ocp-short} cluster. The `local-cluster` is automatically imported in {mce-short} 2.4 and later. See xref:../install_upgrade/adv_config_install.adoc#advanced-config-engine[Advanced configuration] for more information about the `local-cluster`. You can check the status of your hub cluster by running the following command:

+

> [source,terminal]

Reviewer: ```suggestion
[source,bash]
```
I believe the RHACM docs team uses `[source,bash]` instead of `[source,terminal]`. The OCP docs use `[source,terminal]` for most commands and `[source,bash]` for bash scripts.

Author: Noted

----
oc get managedclusters local-cluster
----

* You need a hosting cluster with at least three worker nodes to run the HyperShift Operator.

> * You need to enable Central Infrastructure Management (CIM). For more information, see xref:../cluster_lifecycle/create_infra_env.adoc#enable-cim[Enabling the Central Infrastructure Management service].

Reviewer: ```suggestion
* You need to enable the central infrastructure management service. For more information, see xref:../cluster_lifecycle/create_infra_env.adoc#enable-cim[Enabling the Central Infrastructure Management service].
```

Author: Done

* You need to install the hosted control plane command line interface. See xref:../clusters/hosted_control_planes/hosted_install_cli.adoc[Installing the hosted control plane command line interface].

[#infrastructure-reqs-ibmz]
== IBM Z infrastructure requirements

The Agent platform does not create any infrastructure, but requires the following for infrastructure:

* Agents: An _Agent_ represents a host that is booted with a discovery image, or PXE image and is ready to be provisioned as an {ocp-short} node.

* DNS: The API and Ingress endpoints must be routable.

[#configuring-hcp-bare-metal-ibmz]

> == Configuring hosted control planes on 'x86' bare metal for IBM Z compute nodes

Reviewer: ```suggestion
== Next steps
```
Per recent conversations with @swopebe, we decided to standardize on the "Next steps" heading in cases like this, where we're including links to next steps.

Author: Done

- Each IBM Z system host must be started with the PXE images that is provided by the CIM. After each host starts, it runs an Agent process to discover the details of the host and completes the installation. An Agent custom resource represents each host.

- IBM Z does not support ISO boot. 

- When you create a hosted cluster with the Agent platform, HyperShift installs the Agent Cluster API provider in the hosted control plane namespace.

- When you scale up a node pool, a machine is created. The Cluster API provider finds an Agent that is approved, is passing validations, is not currently in use, and meets the requirements that are specified in the node pool specification. You can monitor the installation of an Agent by checking its status and conditions.

- When you scale down a node pool, Agents are unbound from the corresponding cluster. Before you reuse the clusters, you must boot the clusters by using the PXE image to update the number of nodes.

> [#hosting-service-cluster-configure-prereq]

Reviewer 2: ```suggestion
[#hosting-service-cluster-configure-prereq-ibmz]
```

Author: Same comment as before - Additionally i noticed that the rest of the sub-headings, dont necessarily have IBMz suffixed to the titles.

@SNiemann15

Reviewer 3: Please use the IBM z so that our team when creating links and such can better see where this file belongs. It is not as easy here as it was at IBM where the filenames are in the URLs. Thanks.

> [#configuring-hosting-service-cluster-configure-ibmz]

Reviewer 3: Note for me to pick up here with removing note and labeling tech preview as the rest of the doc.

+
----
oc get managedclusters local-cluster
----

* You need a hosting cluster with at least 3 worker nodes to run the HyperShift Operator.

* You need to enable Central Infrastructure Management (CIM). For more information, see xref:../cluster_lifecycle/create_infra_env.adoc#enable-cim[Enabling the Central Infrastructure Management service].

* You need to link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.9/html/clusters/cluster_mce_overview#hosted-install-cli[install the hosted control plane command line interface (hcp) binary].

[#infrastructure-reqs-bare-metal]
== IBM Z infrastructure requirements

The Agent platform does not create any infrastructure, but it does have the following requirements for infrastructure:

* Agents: An _Agent_ represents a host that is booted with a discovery image or PXE image and is ready to be provisioned as an {ocp-short} node.

* DNS: The API and Ingress endpoints must be routable.

> [#configuring-hcp-bare-metal]

Reviewer 3: we title this configuring, but most of the topic is requirements, then the actual configuring is links to other topics, so let's explore changing this format.
